dahyun1226/app/src/main/java/com/example/android_toy_project_study_2020_mvvm/api/MainActivityModel.kt

```diff
@@ -0,0 +1,56 @@
+package com.example.android_toy_project_study_2020_mvvm.api
+
+import android.app.Activity
+import android.view.View.INVISIBLE
+import android.view.View.VISIBLE
+import com.example.android_toy_project_study_2020_mvvm.data.GithubResponseData
+import com.example.android_toy_project_study_2020_mvvm.recyclerview.ItemAdapter
+import com.example.android_toy_project_study_2020_mvvm.recyclerview.GitItem
+import kotlinx.android.synthetic.main.activity_main.*
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class MainActivityModel (MainActivity: Activity){
+    val mainActivity: Activity = MainActivity
+    fun githubSearch(keyword: String){
+        RetrofitService.getService().requestGithubResponse(keyword = keyword).enqueue(object: Callback<GithubResponseData> {
+            override fun onFailure(call: Call<GithubResponseData>, t: Throwable) {
+                mainActivity.loading.visibility = INVISIBLE
+                mainActivity.errorTextViewRepository.text = "에러발생 : " + t.toString()
+                mainActivity.errorTextViewRepository.visibility = VISIBLE
+                mainActivity.recycler.visibility = INVISIBLE
+            }
+            override fun onResponse(call: Call<GithubResponseData>, response: Response<GithubResponseData>) {
+                if (response.isSuccessful) {
+                    mainActivity.loading.visibility = INVISIBLE
+                    mainActivity.errorTextViewRepository.visibility = INVISIBLE
+                    val githubResponseData = response.body()
+                    mainActivity.loading.visibility = INVISIBLE
+                    val list=ArrayList<GitItem>()
+                    if (githubResponseData != null) {
+                        if (githubResponseData.items.isEmpty()){
+                            mainActivity.errorTextViewRepository.text = "결과 없음!!!"
+                            mainActivity.errorTextViewRepository.visibility = VISIBLE
+                            mainActivity.recycler.visibility = INVISIBLE
+                        } else {
+                            for (i in githubResponseData.items.indices) {
```

Comment:
for문을 사용해도 좋지만 코틀린 컬렉션에는 map() 이라는 오퍼레이터도 제공해줍니다.

```suggestion
  githubResponseData.items.map { //.. }
```

Reply:
사용해보았습니다!

Reply:
커밋 번호를 붙여주세요 ㅠㅜ

Reply:
넵!
690dffe